bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gems/Canopus.java

```diff
@@ -414,48 +271,46 @@ private Area probe3DependentRange(ObsContext ctx, Point2D.Double mid, Tuple2<Dou
             // the center position.  We could just translate to the cwfs star
             // position, but then the width of the rectangle wouldn't be in
             // position to cover the probe 3 range.
-//            xlat = new Point2D.Double(center.getX(), cwfs3.getY());
-            xlat = new Point2D.Double(cwfs3.getX(), cwfs3.getY());
-
-            // Rotate this point to where it should be with the position angle.
-            rotWithPosAngle.transform(xlat, xlat);
-        }
+            final Point2D p = new Point2D.Double(cwfs3.getX(), cwfs3.getY());
+            return rotWithPosAngle.transform(p, p);
+        }).getOrElse(center);
 
         rect.transform(rotWithPosAngle);
         rect.transform(AffineTransform.getTranslateInstance(xlat.getX(), xlat.getY()));
         rect.intersect(p3range);
         return rect;
     }
 
-    public Area probeRange1(ObsContext ctx) {
+    public Area probeRange1(final ObsContext ctx) {
         return probe3DependentRange(ctx, PROBE1_CENTER, PROBE1_DIM);
     }
 
-    public Area probeRange2(ObsContext ctx) {
+    public Area probeRange2(final ObsContext ctx) {
         return probe3DependentRange(ctx, PROBE2_CENTER, PROBE2_DIM);
     }
 
-    public Area probeRange3(ObsContext ctx) {
+    public Area probeRange3(final ObsContext ctx) {
         return offsetIntersection(ctx, ctx.getSciencePositions());
     }
 
     /**
-    * Gets the intersection of the FOV and the offsets.
-    *
-    * @param offsets positions to include in the intersection
-    */
-    public Area offsetIntersection(ObsContext ctx, Set<Offset> offsets) {
+     * Gets the intersection of the FOV and the offsets.
+     *
+     * @param offsets positions to include in the intersection
+     */
+    public Area offsetIntersection(final ObsContext ctx, final Set<Offset> offsets) {
+        // Note: If offsets are ever empty, this will return null, and this is not checked for anywhere.
```

Comment:
This makes me uncomfortable and could be done instead by using `map` and `reduce` over `offsets`, but that gives us an `Optional<Area>`, which makes things more complicated.

There is an assumption that `offsets` will always be non-empty: if it ever is empty, this method returns `null` and will cause an NPE in a couple places.

Reply:
This seems wrong to me too.  If there are no offset positions then we should have the unclipped whole AO port. at the base.

Reply:
Excellent. Thanks. I really didn't know what to do about this. This will clean things up considerably and eliminate the possibility of an NPE here.